src/tir/analysis/calculate_allocated_memory.cc

```diff
@@ -87,6 +87,15 @@ TVM_REGISTER_GLOBAL("tir.analysis.calculate_allocated_bytes").set_body_typed([](
   return CalculateAllocatedBytes(func);
 });
 
+bool VerifyVTCMLimit(const PrimFunc& func, Integer limit) {
```

Comment:
looks like `transform::VerifyVTCMLimit` also uses this logic. Can we reuse this function inside that one?

Reply:
Yeah I wanted to do that, but the pass variant needs `vtcm_allocated` in its error message, while this one wants to return only true / false. Refactoring is possible but I thought just duplicating might be cleaner in this case.